Sources/BSON/BSONDocumentIterator.swift

```diff
@@ -108,6 +97,52 @@ public struct BSONDocumentIterator: IteratorProtocol {
             }
         }
     }
+
+    // uses an iterator to copy (key, value) pairs of the provided document from range [startIndex, endIndex) into a new
+    // document. starts at the startIndex-th pair and ends at the end of the document or the (endIndex-1)th index,
+    // whichever comes first.
+    internal static func subsequence(
+        of doc: BSONDocument,
+        startIndex: Int = 0,
+        endIndex: Int = Int.max
+    ) -> BSONDocument {
+        // TODO(SWIFT-911): Improve performance
```

Comment:
the format of our TODOs is typically `TODO: SWIFT-XXX <reason>`. We should probably stick to this for easy grepping in the future.